src/python/pants/core/goals/generate_lockfiles.py

```diff
@@ -423,13 +603,35 @@ async def generate_lockfiles_goal(
         for req in all_requests
     )
 
+    # Generate diffs before writing the new files to the workspace while the previous version is
```

Comment:
Why have this logic here? Why not add the diff result into the GenerateLockfileResult in the backend? Then we wouldn't need generate_diff_cls etc. I think?

Reply:
Thanks, that is worth considering.

Reply:
Refactored so we now pass a `diff: bool` with the request, and return a `diff: LockfileGenerateDiffResult|None` with the response.